Tasks/Kubernetes/task.json

```diff
@@ -112,52 +147,53 @@
                 "Container Registry": "Container Registry"
             },
             "helpMarkDown": "Select a Container registry type.",
-            "groupName": "registry"
+            "visibleRule": "secretType = dockerRegistry",
+            "groupName": "secrets"
         },
         {
             "name": "dockerRegistryEndpoint",
             "aliases": ["dockerRegistryConnection"],
             "type": "connectedService:dockerregistry",
             "label": "Docker Registry Connection",
             "helpMarkDown": "Select a Docker registry connection. Required for commands that need to authenticate with a registry.",
-            "visibleRule": "containerRegistryType = Container Registry",
-            "groupName": "registry"
+            "visibleRule": "secretType = dockerRegistry && containerRegistryType = Container Registry",
+            "groupName": "secrets"
         },
         {
             "name": "azureSubscriptionEndpoint",
             "aliases": ["azureSubscription"],
             "type": "connectedService:AzureRM",
             "label": "Azure subscription",
             "helpMarkDown": "Select an Azure subscription, which has your Azure Container Registry.",
-            "visibleRule": "containerRegistryType = Azure Container Registry",
+            "visibleRule": "secretType = dockerRegistry && containerRegistryType = Azure Container Registry",
             "defaultValue": "",
-            "groupName": "registry"
+            "groupName": "secrets"
```

Comment:
Changing group of an old input will not break the old task right?. Have we verified it.

Reply:
Yes, i have verified. The task will run the same